File: Algorithms/Python/128 Longest Consecutive Sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        if len(nums) == 0:
            return 0

        nums_set = set(nums)
        ans = 1
        for n in nums:
            if n - 1 not in nums_set:
                seq_len = 1
                cur = n + 1
                while cur in nums_set:
                    cur += 1
                    seq_len += 1
                ans = max(ans, seq_len) 
        return ans

File: Algorithms/CPP/128 Longest Consecutive Sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        if (nums.size() == 0) {
            return 0;
        }

        std::unordered_set<int> nums_set(nums.begin(), nums.end());
        int ans = 1;
        for (int i = 0; i < nums.size(); ++i) {
            if (nums_set.find(nums[i] - 1) == nums_set.end()) {
                int cur = nums[i] + 1;
                int seq_len = 1;
                while (nums_set.find(cur) != nums_set.end()) {
                    ++cur;
                    ++seq_len;
                }
                ans = max(ans, seq_len);
            }
        }
        return ans;
    }
};
